File: Java/src/codewars/Beggars.java
// https://www.codewars.com/kata/59590976838112bfea0000fa/java

package codewars;

public class Beggars {
    public static int[] beggars(int[] values, int n) {
        var arr = new int[n];
        for (int i = 0; i < n; i++) {
            var res = 0;
            for (int j = i; j < values.length; j += n) {
                res += values[j];
            }
            arr[i] = res;
        }

        return arr;
    }
}

File: Swift/Sources/Codewars/Beggars.swift
// https://www.codewars.com/kata/59590976838112bfea0000fa/swift

struct Beggars
{
	static func beggars(_ values: [Int], _ n: Int) -> [Int]
	{
		var arr: [Int] = []
		for i in stride(from: 0, to: n, by: 1) 
		{
			var res = 0
			for j in stride(from: i, to: values.count, by: n)
			{
				res += values[j]
			}
			arr.append(res)
		}
		return arr
	}
}
